src/Microsoft.TemplateEngine.Utils/PhysicalFileSystem.cs

```diff
@@ -101,5 +101,18 @@ public void SetLastWriteTimeUtc(string file, DateTime lastWriteTimeUtc)
         {
             File.SetLastWriteTimeUtc(file, lastWriteTimeUtc);
         }
+
+        public IDisposable WatchFileChanges(string filepath, FileSystemEventHandler fileChanged)
+        {
```

Comment:
This looks like reinventing of Reactive behaviors already available in the extensions.